workflow/artifacts/git/git_test.go

```diff
@@ -1,105 +1,128 @@
 	assert.Error(t, err)
 }
 
-func TestGitArtifactDriverLoad_HTTPS(t *testing.T) {
-	for _, tt := range []struct {
-		url string
-	}{
-		{"https://github.com/argoproj/empty.git"},
-	} {
-		if os.Getenv("GITHUB_TOKEN") == "" {
-			t.Skip("not running an GITHUB_TOKEN not set")
-		}
-		_ = os.Remove("git-ask-pass.sh")
-		tmp := t.TempDir()
-		driver := &ArtifactDriver{Username: os.Getenv("GITHUB_TOKEN")}
-		assert.NotEmpty(t, driver.Username)
-		err := driver.Load(&wfv1.Artifact{
-			ArtifactLocation: wfv1.ArtifactLocation{
-				Git: &wfv1.GitArtifact{
-					Repo:     tt.url,
-					Fetch:    []string{"+refs/heads/*:refs/remotes/origin/*"},
-					Revision: "HEAD",
-					Depth:    &d,
-				},
-			},
-		}, tmp)
-		assert.NoError(t, err)
-		println(tmp)
-	}
-}
-
-func TestGitArtifactDriverLoad_SSL(t *testing.T) {
-	t.SkipNow()
-	for _, tt := range []struct {
-		name     string
-		insecure bool
-		url      string
-	}{
-		{"Insecure", true, "https://github.com/argoproj/argo-workflows.git"},
-		{"Secure", false, "https://github.com/argoproj/argo-workflows.git"},
-		{"Insecure", true, "https://github.com/argoproj/empty.git"},
-		{"Secure", false, "https://github.com/argoproj/empty.git"},
-	} {
-		t.Run(tt.name, func(t *testing.T) {
-			_ = os.Remove("git-ask-pass.sh")
-			key := os.Getenv("HOME") + "/.ssh/id_rsa"
-			data, err := ioutil.ReadFile(key)
-			if err != nil && os.IsNotExist(err) {
-				t.Skip(key + " does not exist")
+func TestGitArtifactDriver_Load(t *testing.T) {
+	t.Run("EmptyRepo", func(t *testing.T) {
+		driver := &ArtifactDriver{}
+		assert.NoError(t, load(driver, &wfv1.GitArtifact{Repo: "https://github.com/argoproj-labs/empty-test-repo.git"}))
+		assert.DirExists(t, path)
+	})
+	t.Run("PrivateRepo", func(t *testing.T) {
+		t.Run("SSH", func(t *testing.T) {
+			if os.Getenv("CI") == "true" {
+				t.SkipNow()
 			}
+			privateKey, err := os.ReadFile(homedir.HomeDir() + "/.ssh/id_rsa")
 			assert.NoError(t, err)
-			tmp := t.TempDir()
-			println(tmp)
-			driver := &ArtifactDriver{SSHPrivateKey: string(data)}
-			err = driver.Load(&wfv1.Artifact{
-				ArtifactLocation: wfv1.ArtifactLocation{
-					Git: &wfv1.GitArtifact{
-						Repo:                  tt.url,
-						Fetch:                 []string{"+refs/heads/*:refs/remotes/origin/*"},
-						Revision:              "HEAD",
-						InsecureIgnoreHostKey: tt.insecure,
-						Depth:                 &d,
-					},
-				},
-			}, tmp)
-			assert.NoError(t, err)
+			driver := &ArtifactDriver{SSHPrivateKey: string(privateKey)}
+			assert.NoError(t, load(driver, &wfv1.GitArtifact{Repo: "git@github.com:argoproj-labs/private-test-repo.git"}))
+			assert.FileExists(t, path+"/README.md")
 		})
-	}
+		t.Run("HTTPS", func(t *testing.T) {
+			token := os.Getenv("PERSONAL_ACCESS_TOKEN")
+			if token == "" {
+				t.SkipNow()
+			}
+			driver := &ArtifactDriver{Username: "alexec", Password: token}
+			assert.NoError(t, load(driver, &wfv1.GitArtifact{Repo: "https://github.com/argoproj-labs/private-test-repo.git"}))
```

Comment:
This test requires the existence of this private repo and requires a personal access token. Is there a better way?

Reply:
this only runs locally